frontera/core/models.py

```diff
@@ -23,10 +24,7 @@ def __init__(self, url, method='GET', headers=None, cookies=None, meta=None, bod
         :param dict cookies: dictionary of cookies to attach to this request.
         :param dict meta: dictionary that contains arbitrary metadata for this request.
         """
-        self._url = url
-        self._method = str(method).upper()
-        self._url = to_native_str(url, encoding)
-        self._encoding = encoding
+        self._url = safe_url_string(url, encoding)
```

Comment:
Frontera is designed to work without Scrapy too. So, we need this URL transformation, the question is how to make it cheap. That's why I propose to optimize it later.